d2common/data_dictionary.go

```diff
@@ -20,16 +20,16 @@ func LoadDataDictionary(text string) *DataDictionary {
 	for i, fieldName := range fileNames {
 		result.FieldNameLookup[fieldName] = i
 	}
-	result.Data = make([][]string, len(lines)-1)
```

Comment:
Why did you make this change? Won't repeatedly calling append() add a bunch of stuff that the gc has to free after this is done?

Reply:
You can at least allocate the full array and leave the slice empty.  But it looks like this is to fix an off by one error: it allocates len - 1 but since the loop started at index 1 it left an empty record.  Imo change it to len - 2 if we want to be kind to GC or `make([][]string, 0, len(lines) - 2` if you want to be defensive.

Reply:
Yes, it was to fix an OBO error. I've made the edit, thank you.